core/controllers/base.py

```diff
@@ -571,6 +571,8 @@ def _render_exception_json_or_html(self, return_type, values):
                     'error-iframed.mainpage.html', iframe_restriction=None)
             elif values['status_code'] == 503:
                 self.render_template('maintenance-page.mainpage.html')
+            elif values['status_code'] == 404:
+                self.render_template('oppia-root.mainpage.html')
```

Comment:
What does this new page look like when it occurs? Could you add a screenshot to provide more context for the changed 404 experience?

Reply:
https://user-images.githubusercontent.com/24855641/126206315-86620979-647f-4c8c-a779-3568fec17560.mp4

I recommend going through the proposal. https://docs.google.com/document/d/17LSN93KaJmEZ7k5Sjli_wfz4TEsiHMPEjcXQeFtLAL4/edit?usp=sharing

Reply:
The end outcome of my project will look like this:
switching in between pages will be a lot faster.

https://user-images.githubusercontent.com/24855641/126447439-d75e49c1-5426-4cf5-b1e1-b65749b049c4.mp4

Reply:
The error page will remain same but it will served using the unified entry point `oppia-root.mainpage.html`.

Reply:
I think you could add some explanation, in the form of a clear comment, about why a 404 is treated differently here from a 400, 401, 500, etc.

Reply:
done

Reply:
Thanks for the context @ashutoshc8101. The comment is especially what I was looking for here.